server/src/main/java/org/opensearch/bootstrap/Security.java

```diff
@@ -416,6 +423,17 @@ private static void addSocketPermissionForHttp(final Permissions policy, final S
         addSocketPermissionForPortRange(policy, httpRange);
     }
 
+    /**
+     * Add dynamic {@link SocketPermission} based on gRPC settings.
+     *
+     * @param policy the {@link Permissions} instance to apply the dynamic {@link SocketPermission}s to.
+     * @param settings the {@link Settings} instance to read the gRPC settings from
+     */
+    private static void addSocketPermissionForGrpc(final Permissions policy, final Settings settings) {
+        final String grpcRange = GrpcTransportSettings.SETTING_GRPC_PORT.get(settings).getPortRangeString();
```

Comment:
This looks wrong, the GRCP plugin policy has to include the permission, it may need placeholders support for port ranges though (but it could be implemented).

Reply:
I think that was the hangup? The policy must be hard-coded but the specific port needs to be user-configurable.

Reply:
We do allow placeholders (for code bases fe), we could implement the placeholders for ports (or open up some limited subset fe)

Reply:
We'd need to get the specific value from the configuration at runtime at startup. Do we have any existing examples like that?

Reply:
So something like `addSocketPermissionForPluginTransports` and allow each plugin to consume from the available pool?

Reply:
> We'd need to get the specific value from the configuration at runtime at startup. Do we have any existing examples like that?

Yes, we do, check please `Security::addSocketPermissionForTransportProfiles`